src/frequenz/sdk/actor/power_distributing/_battery_status.py

```diff
@@ -223,8 +224,80 @@ async def stop(self) -> None:
         """Stop tracking battery status."""
         await cancel_and_await(self._task)
 
-        if self._select is not None:
-            await self._select.stop()
+    def _handle_status_battery(
```

Comment:
Very nice refactoring here with one method to handle each receiver. The diff ended up a bit hard to read though, so in the future I would recommend to split the refactoring to a separate commit and in this commit try to make the code structure as similar as it originally was.

In this case keeping the `_update_status` method and pass the `selected` object instead.